handler.go

```diff
@@ -108,11 +112,29 @@ func (r requestHandler) handleStreamRequest() {
 	}
 
 	if r.shouldMakeInitialRequest() {
-		err = makeStreamRequest(r.transport, streamReq, streamSerializer, requestSupplier, responseHandler)
+		err = makeStreamRequest(r.transport, streamReq, r.serializer, requestSupplier, responseHandler)
 		if err != nil {
 			r.out.Fatalf("%v\n", err)
 		}
+	} else {
+		// read the request messages explicitly as there was no initial request
+		for {
+			_, err := requestSupplier()
+			if err == io.EOF {
+				break
+			}
+			if err != nil {
+				r.out.Fatalf("%v\n", err)
+			}
+		}
```

Comment:
Why do we need this? can we simply gives `requestSupplier` to `benchmarkMethod` ?

Reply:
benchmarkMethod is shared between multiple go-routines, the requests from decoder can be read only once hence this can't be shared directly. Also, requestSupplier isn't safe to be accessed in parallel. That's why the requests are accumulated and shared as a slice of request bytes

Reply:
I see, is there any chance the benchmarkMethod struct takes care of this? since we will have a one type for streaming and one for unary, the streaming one should be able to do this no?

Reply:
This wouldn't work cleanly as benchmark could be coupled with a warm-up request. If there is a warm-up request, then request reader would have hit EOF meaning we cannot pass it to benchmarker anymore.

Reply:
I see, this logic is a bit hard to follow and maintain IMO. The issues is that the code to read values from the streamer is split into 3 different parts (here, line:92 of this file, and in `benchmark_streaming.go`).

What do you think about the following proposals:

Instead of having the function `nextBodyFn := func() ([]byte, error) {}` (see https://github.com/yarpc/yab/pull/317/files#diff-0eb779b9e49d8e44b0f36923fdb8d87d5ee024f886eefc45deec4ec88380a087R91), we add a new function in handler.go:

```
func nextStreamingBodyFunction (reader, opts) func() ([]byte, error) {
   if opts == "interactive" {
      return func() ([]byte, error) {
         // read with streamMsgReader.NextBody() until eof 
      } 
   }

   // not interactive, read all of the input and create an array of streamRequestMessages[]

    // return the function
    return func() ([]byte, error) {
       return m.streamRequestMessages[streamRequestIdx]
    }
}
``` 

By having this, you would only need to call the method `nextStreamingBodyFunction` once in the handler. You could then give it to the benchmark one.

Reply:
I agree, the logic was spread through the streaming handler. I have split request handler and response handler into separate functions. This must simplify the logic
https://github.com/yarpc/yab/blob/3a22bcdb6f295ad5895ea197dc91d7f516e7244d/handler.go#L140